Justfile

```
@@ -141,6 +145,25 @@ test-rust-gdb-debugging target=default-target features="":
 test-rust-crashdump target=default-target features="":
     cargo test --profile={{ if target == "debug" { "dev" } else { target } }} {{ if features =="" {'--features crashdump'} else { "--features crashdump," + features } }} -- test_crashdump
 
+# rust test for tracing
+test-rust-tracing target=default-target features="":
+    # Run tests for the tracing guest and macro
+    cargo test -p hyperlight-guest-tracing --profile={{ if target == "debug" { "dev" } else { target } }}
+    cargo test -p hyperlight-guest-tracing-macro --profile={{ if target == "debug" { "dev" } else { target } }}
+
+    # Prepare the tracing guest for testing
+    just build-rust-guests {{ target }} trace_guest
+    just move-rust-guests {{ target }}
+    # Run hello-world example with tracing enabled to get the trace output
+    # Capture the trace file path and print use it afterwards to run cargo run -p trace_dump
+    cargo run --profile={{ if target == "debug" { "dev" } else { target } }} --example hello-world --features {{ if features =="" {'trace_guest'} else { "trace_guest," + features } }} \
+        | sed -n 's/.*Creating trace file at: \(.*\)/\1/p' \
```

Should this be checking that the dumped frames match what is expected or anything?

---

A simple check to verify at least `hyperlight_main` got called could be worth doing, although if anything fails, the example should report an error, so the check might be redundant.

---

I'll implement this in the improvement ticket also.